orbitdeterminator/kep_determination/positional_observation_reporting.py

```diff
@@ -249,4 +249,78 @@ def load_iod_data(fname):
     iod["decMM"] = decMM
     iod["decmmm"] = decmmm
 
-    return iod
\ No newline at end of file
+    return iod
+
+##uk format detection
+
+def get_uk_names():
+    uk_names = ['launchyear', 'seqno', 'pieceno',
+                 'siteno', 'yr', 'month','date',
+                 'hr', 'min', 'sec', 'msec', 'timeN', 'timeD',
+                 'timestd', 'positionformatcode',
+                 'obsra', 'obsdec', 'pacc', 'epoch',
+                 'range', 'racc', 'bvismag', 'fvismag', 'flashperiod','remarks']
+
+    return uk_names
+
+def get_uk_lines(fname):
+    # dt is the dtype for UK-formatted text files
+    dt = 'i8, i8, i8,' \
+         ' i8, i8, i8, i8,' \
+         ' i8, i8, i8, i8, i8, i8,' \
+         ' i8, i8,' \
+         ' S8, S8, i8, i8,' \
+         ' i8, i8, S3, S3, i8, S1'
+
+    # iod_names correspond to the dtype names of each field
+    uk_names = get_uk_names()
+
+    # iod_delims corresponds to the delimiter for cutting the right variable from each input string
+    uk_delims = [2, 3, 2,
+                  4, 2, 2, 2,
+                  2, 2, 2, 4, 1, 4,
+                  1, 1,
+                  8, 8, 4, 1,
+                  8, 5, 3, 3, 5, 1]
+
+    uk_input_lines = np.genfromtxt(fname, dtype=dt, names=uk_names, delimiter=uk_delims, autostrip=True)
+
+    return uk_input_lines
+
+
+
+
+def check_uk_format(fname):
+
+    uk_input_lines = get_uk_lines(fname)
+
+    number_of_lines = len(uk_input_lines["launchyear"])
+    confirmed_lines = 0
+
+    for line in range(number_of_lines):
+
+        launchyear = uk_input_lines["launchyear"][line]
+        remarks = uk_input_lines["remarks"][line]
+        fvismag = uk_input_lines["fvismag"][line]
+        epoch = uk_input_lines["epoch"][line]
+        tstd=uk_input_lines["timestd"][line]
+
+        if len(str(launchyear)) == 2 and \
+                any(remarks.decode('ascii') == x for x in ["S","I","R","F","X","E"]) and \
```

Comment:
you are refering to http://www.satobs.org/position/UKformat.html and the "Col      80: Remarks." I think. I don't fully understand yet, if this is always required to be in the report string. it is not stated as clearly as in IOD format, but "comments" sounds for me like this is not mandatory.
can you confirm, that it is a required input?

Reply:
Yes I will remove the condition on remarks as it is mentioned that it can also be left blank.